lib/spack/spack/test/util/util_url.py

```diff
@@ -45,155 +47,52 @@ def test_relative_path_to_file_url(tmpdir):
         assert os.path.samefile(roundtrip, path)
 
 
-def test_url_join_local_paths():
-    # Resolve local link against page URL
-
-    # wrong:
-    assert (
-        url_util.join("s3://bucket/index.html", "../other-bucket/document.txt")
-        == "s3://bucket/other-bucket/document.txt"
-    )
-
-    # correct - need to specify resolve_href=True:
-    assert (
-        url_util.join("s3://bucket/index.html", "../other-bucket/document.txt", resolve_href=True)
-        == "s3://other-bucket/document.txt"
-    )
-
-    # same as above: make sure several components are joined together correctly
-    assert (
-        url_util.join(
-            # with resolve_href=True, first arg is the base url; can not be
-            # broken up
-            "s3://bucket/index.html",
-            # with resolve_href=True, remaining arguments are the components of
-            # the local href that needs to be resolved
-            "..",
-            "other-bucket",
-            "document.txt",
-            resolve_href=True,
-        )
-        == "s3://other-bucket/document.txt"
-    )
-
-    # Append local path components to prefix URL
-
-    # wrong:
-    assert (
-        url_util.join("https://mirror.spack.io/build_cache", "my-package", resolve_href=True)
-        == "https://mirror.spack.io/my-package"
-    )
-
-    # correct - Need to specify resolve_href=False:
-    assert (
-        url_util.join("https://mirror.spack.io/build_cache", "my-package", resolve_href=False)
-        == "https://mirror.spack.io/build_cache/my-package"
-    )
-
-    # same as above; make sure resolve_href=False is default
-    assert (
-        url_util.join("https://mirror.spack.io/build_cache", "my-package")
-        == "https://mirror.spack.io/build_cache/my-package"
-    )
-
-    # same as above: make sure several components are joined together correctly
-    assert (
-        url_util.join(
-            # with resolve_href=False, first arg is just a prefix. No
-            # resolution is done.  So, there should be no difference between
-            # join('/a/b/c', 'd/e'),
-            # join('/a/b', 'c', 'd/e'),
-            # join('/a', 'b/c', 'd', 'e'), etc.
-            "https://mirror.spack.io",
-            "build_cache",
-            "my-package",
-        )
-        == "https://mirror.spack.io/build_cache/my-package"
-    )
-
-    # For s3:// URLs, the "netloc" (bucket) is considered part of the path.
-    # Make sure join() can cross bucket boundaries in this case.
-    args = ["s3://bucket/a/b", "new-bucket", "c"]
-    assert url_util.join(*args) == "s3://bucket/a/b/new-bucket/c"
-
-    args.insert(1, "..")
-    assert url_util.join(*args) == "s3://bucket/a/new-bucket/c"
-
-    args.insert(1, "..")
-    assert url_util.join(*args) == "s3://bucket/new-bucket/c"
```

Comment:
The tests don't really apply because the signature has changed from join(*args) to join(x, y).

The new tests are pretty clear in that http/s3/gs/file all work uniformly, the old tests were not.